src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -466,6 +480,44 @@ export class StudyViewPageStore
             )
         );
 
+        this.reactionDisposers.push(
```

Comment:
i suspect that it is the reactions which may be causing the charts to re-render whenever hesitateUpdate changes.